dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/ajax/ContentletAjax.java

```diff
@@ -851,209 +853,219 @@ else if( fieldbcontentname.startsWith("date") ){
+						expiredInodes.add(con.getInode()); // it is unpublished and can't be manualy published
 				}
-				Field field = (Field) fieldsMapping.get(fieldContentlet);
-				if (UtilMethods.isSet(fieldValue) && field.getFieldType().equals(Field.FieldType.DATE.toString()) ||
-						UtilMethods.isSet(fieldValue) && field.getFieldType().equals(Field.FieldType.TIME.toString()) ||
-						UtilMethods.isSet(fieldValue) && field.getFieldType().equals(Field.FieldType.DATE_TIME.toString())) {
-					try {
-						Date date = DateUtil.convertDate(fieldValue, new String[] { "yyyy-MM-dd HH:mm:ss", "E MMM dd HH:mm:ss z yyyy" });
-						if (field.getFieldType().equals(Field.FieldType.DATE.toString()))
-							fieldValue = UtilMethods.dateToHTMLDate(date);
-						if (field.getFieldType().equals(Field.FieldType.TIME.toString()))
-							fieldValue = UtilMethods.dateToHTMLTime(date);
-						if (field.getFieldType().equals(Field.FieldType.DATE_TIME.toString()))
-							fieldValue = UtilMethods.dateToHTMLDate(date) + " " + UtilMethods.dateToHTMLTime(date);
-					} catch (java.text.ParseException e) {
-						Logger.error(ContentletAjax.class, e.getMessage(), e);
-						throw new DotRuntimeException(e.getMessage(), e);
+
+				searchResult = new HashMap<String, String>();
+				Structure s = CacheLocator.getContentTypeCache().getStructureByInode(con.getStructureInode());
+
+				for (String fieldContentlet : fieldsMapping.keySet()) {
+					String fieldValue = null;
+					if (con.getMap() != null && con.getMap().get(fieldContentlet) != null) {
+						fieldValue = (con.getMap().get(fieldContentlet)).toString();
+					}
+					Field field = (Field) fieldsMapping.get(fieldContentlet);
+					if (UtilMethods.isSet(fieldValue) && field.getFieldType().equals(Field.FieldType.DATE.toString()) ||
+							UtilMethods.isSet(fieldValue) && field.getFieldType().equals(Field.FieldType.TIME.toString()) ||
+							UtilMethods.isSet(fieldValue) && field.getFieldType().equals(Field.FieldType.DATE_TIME.toString())) {
+						try {
+							Date date = DateUtil.convertDate(fieldValue, new String[]{"yyyy-MM-dd HH:mm:ss", "E MMM dd HH:mm:ss z yyyy"});
+							if (field.getFieldType().equals(Field.FieldType.DATE.toString()))
+								fieldValue = UtilMethods.dateToHTMLDate(date);
+							if (field.getFieldType().equals(Field.FieldType.TIME.toString()))
+								fieldValue = UtilMethods.dateToHTMLTime(date);
+							if (field.getFieldType().equals(Field.FieldType.DATE_TIME.toString()))
+								fieldValue = UtilMethods.dateToHTMLDate(date) + " " + UtilMethods.dateToHTMLTime(date);
+						} catch (java.text.ParseException e) {
+							Logger.error(ContentletAjax.class, e.getMessage(), e);
+							throw new DotRuntimeException(e.getMessage(), e);
+						}
+					} else if (field.getFieldType().equals(Field.FieldType.CHECKBOX.toString()) || field.getFieldType().equals(Field.FieldType.MULTI_SELECT.toString())) {
+						if (UtilMethods.isSet(fieldValue))

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)

Reply:
Legacy code, do not wanna chage